### PR TITLE
Poao tilgang

### PR DESCRIPTION
**Pull request by dragutin-nav**

### CODE REVIEW

File: src/main/java/no/nav/pto/veilarbportefolje/aktiviteter/AktivitetService.java [hunk 1]
```diff
@@ -27,16 +30,42 @@ public class AktivitetService extends KafkaCommonConsumerService<KafkaAktivitetM
     private final OppfolgingsbrukerRepositoryV3 oppfolgingsbrukerRepository;
     private final SisteEndringService sisteEndringService;
     private final OpensearchIndexer opensearchIndexer;
+    private final TiltakService tiltakService;
+    private final UnleashService unleashService;
 
     public void behandleKafkaMeldingLogikk(KafkaAktivitetMelding aktivitetData) {
         AktorId aktorId = AktorId.of(aktivitetData.getAktorId());
         sisteEndringService.behandleAktivitet(aktivitetData);
+
```
Comment (OdaDahlen) [inline src/main/java/no/nav/pto/veilarbportefolje/aktiviteter/AktivitetService.java:39]: These changes should not be here, I guess you have to pull from dev and merge it into your PR

File: src/main/java/no/nav/pto/veilarbportefolje/auth/AuthService.java [hunk 2]
```diff
@@ -58,12 +62,27 @@ public void tilgangTilEnhet(String enhet) {
     }
 
     public boolean harVeilederTilgangTilEnhet(String veilederId, String enhet) {
-        return tryCacheFirst(harVeilederTilgangTilEnhetCache, new VeilederPaEnhet(veilederId, enhet),
+        Boolean abacResponse = tryCacheFirst(harVeilederTilgangTilEnhetCache, new VeilederPaEnhet(veilederId, enhet),
                 () -> veilarbPep.harVeilederTilgangTilEnhet(NavIdent.of(veilederId), EnhetId.of(enhet)));
+
+        if (FeatureToggle.brukPoaoTilgang(unleashService)) {
+            Decision decision = poaoTilgangWrapper.harVeilederTilgangTilEnhet(EnhetId.of(enhet));
+            if (decision.isPermit() != abacResponse) {
```
Comment (JulieHillRoa) [inline src/main/java/no/nav/pto/veilarbportefolje/auth/AuthService.java:70]: Alexander Gård has actually added this in poao-tilgang:
https://github.com/navikt/poao-tilgang/blob/main/core/src/main/kotlin/no/nav/poao_tilgang/core/policy/impl/NavAnsattTilgangTilNavEnhetPolicyImpl.kt

File: src/main/java/no/nav/pto/veilarbportefolje/auth/AuthService.java [hunk 3]
```diff
@@ -58,12 +62,27 @@ public void tilgangTilEnhet(String enhet) {
...
+        }
+        return abacResponse;
     }
 
     public void tilgangTilBruker(String fnr) {
-        AuthUtils.test("tilgangTilBruker", fnr, veilarbPep.harTilgangTilPerson(getInnloggetBrukerToken(), ActionId.READ, Fnr.of(fnr)));
+        boolean abacResponse = veilarbPep.harTilgangTilPerson(getInnloggetBrukerToken(), ActionId.READ, Fnr.of(fnr));
+        if (FeatureToggle.brukPoaoTilgang(unleashService)) {
+            Decision decision = poaoTilgangWrapper.harTilgangTilPerson(Fnr.of(fnr));
+            if (decision.isPermit() != abacResponse) {
```
Comment (JulieHillRoa) [inline src/main/java/no/nav/pto/veilarbportefolje/auth/AuthService.java:81]: Alexander Gård has added this in poao-tilgang:
https://github.com/navikt/poao-tilgang/blob/main/core/src/main/kotlin/no/nav/poao_tilgang/core/policy/impl/NavAnsattTilgangTilEksternBrukerPolicyImpl.kt#L26


We only need to add the logging in "new" policies (i.e the ones he hasn't already fixed haha) 🎉

File: src/main/java/no/nav/pto/veilarbportefolje/auth/AuthService.java (hunk 2 above)
Comment (dragutin-nav) [reply src/main/java/no/nav/pto/veilarbportefolje/auth/AuthService.java:70]: Yep yep, skal fikse..

File: src/main/java/no/nav/pto/veilarbportefolje/auth/AuthService.java [hunk 4]
```diff
@@ -74,37 +93,70 @@ public List<Bruker> sensurerBrukere(List<Bruker> brukere) {
...
             return AuthUtils.fjernKonfidensiellInfo(bruker);
         }
         return bruker;
+    }
 
+    private boolean harVeilederTilgangTilKode6(NavIdent veilederIdent) {
+        boolean abacResponse = veilarbPep.harVeilederTilgangTilKode6(veilederIdent);
+        if (FeatureToggle.brukPoaoTilgang(unleashService)) {
+            Decision decision = poaoTilgangWrapper.harVeilederTilgangTilKode6();
+            if (decision.isPermit() != abacResponse) {
```
[the remark this answers is not in the document]
Comment (dragutin-nav) [reply src/main/java/no/nav/pto/veilarbportefolje/auth/AuthService.java:118]: seems like we are not using abac in poao-tilgang for this function, so its not possible to log differences there (ref. https://github.com/navikt/poao-tilgang/blob/main/core/src/main/kotlin/no/nav/poao_tilgang/core/policy/impl/NavAnsattBehandleStrengtFortroligBrukerePolicyImpl.kt)

**Review by JulieHillRoa — APPROVED** [review]

**Pull request merged**